packages/core/src/validation.ts

```diff
@@ -305,7 +305,7 @@ export function validateString(str: string, label?: string): ValidateReturn {
 }
 
 export function validateSetChainOptions(data: {
-  chainId: ChainId
+  chainId: ChainId | DecimalChainId
```

Comment:
if change the types to `type ChainId = number | string` then we could have just `ChainId` here